pkg/app/pagewriter/pagewriter.go

```diff
@@ -3,14 +3,16 @@ package pagewriter
 import (
 	"fmt"
 	"net/http"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/tenant/types"
```

Comment:
Do we need another concept, a tenant? Why not extend the provider interface?

Reply:
responded [here](https://github.com/oauth2-proxy/oauth2-proxy/pull/1923#issuecomment-1369665582),

Reply:
the tenant struct and loader interfaces have been removed. Only a concep of tenant-id remains now. We extract tenant-id from the incoming request. We could have replaced tenant-id as well and only captured provider-id from the request. But in order to make the whole module multi-tenant, we must have the tenant-concept in some form. Since there are other components besides providers that need to be modified to make them multi-tenant. For example, basicAuth validator, email domain validator and SkipJwtBearerTokens option.